iroh-net/src/magicsock.rs

```diff
@@ -1809,22 +1829,29 @@ impl Actor {
                         self.refresh_direct_addrs(reason).await;
                     }
                 }
-                Some(is_major) = link_change_r.recv() => {
+                is_major = link_change_r.recv(), if !link_change_closed => {
+                    let Some(is_major) = is_major else {
+                        trace!("tick: link change receiver closed");
+                        inc!(Metrics, actor_tick_other);
+
+                        link_change_closed = true;
+                        continue;
```

Comment:
Here I also think this is a critical error.  Clearly it doesn't happen much, so maybe let's just log this on error level for now and carry on.  Long term the link monitor (or whatever) can make sure it is supervised and doesn't die on us like that.

Reply:
I think technically things can continue fine if the network monitor were to close as long as the network doesn't change, or even if it does as long as  `Endpoint::network_change` is used.

But I do agree that it should probably be logged at a higher level - I'll change it to `warn`